providers/implementations/keymgmt/ecx_kmgmt.c

```diff
@@ -670,6 +674,104 @@ void *ecx_load(const void *reference, size_t reference_sz)
     return NULL;
 }
 
+static int ecx_key_public_check(const ECX_KEY *ecx, size_t len)
+{
+    return ecx->haspubkey && len == ecx->keylen;
+}
+
+static int ecx_key_private_check(const ECX_KEY *ecx, int type, size_t len)
+{
+    const unsigned char *priv = ecx->privkey;
+
+    if (priv == NULL || len != ecx->keylen)
```

Comment:
We could even consider an `!ossl_assert` on the `len != ecx->keylen` case. This should never happen.

Reply:
Done..